tests/compute/test_csrmm.py

```diff
@@ -11,51 +12,199 @@ def _random_simple_graph(idtype, dtype, ctx, M, N, max_nnz, srctype, dsttype, et
     a = ssp.csr_matrix((val, (src, dst)), shape=(M, N))
     a.sum_duplicates()
     a = a.tocoo()
+    # shuffle edges
+    perm = np.random.permutation(a.nnz)
+    row = a.row[perm]
+    col = a.col[perm]
+    val = a.data[perm]
+    a = ssp.csr_matrix((val, (row, col)), shape=(M, N))
+
     A = dgl.heterograph(
-        {('A', 'AB', 'B'): (
-            F.copy_to(F.tensor(a.row, dtype=idtype), ctx),
-            F.copy_to(F.tensor(a.col, dtype=idtype), ctx))},
-        num_nodes_dict={'A': a.shape[0], 'B': a.shape[1]})
-    A.edata['w'] = F.copy_to(F.tensor(a.data, dtype=dtype), ctx)
+        {(srctype, etype, dsttype): (
+            F.copy_to(F.tensor(row, dtype=idtype), ctx),
+            F.copy_to(F.tensor(col, dtype=idtype), ctx))},
+        num_nodes_dict={srctype: a.shape[0], dsttype: a.shape[1]})
+    A.edata['w'] = F.copy_to(F.tensor(val, dtype=dtype), ctx)
     return a, A
 
 @parametrize_dtype
-def test_csrmm(idtype):
-    for dtype in [F.float32, F.float64]:
-        a, A = _random_simple_graph(idtype, dtype, F.ctx(), 500, 600, 9000, 'A', 'B', 'AB')
-        b, B = _random_simple_graph(idtype, dtype, F.ctx(), 600, 700, 9000, 'B', 'C', 'BC')
-        C, C_weights = dgl.sparse.csrmm(A._graph, A.edata['w'], B._graph, B.edata['w'], 2)
-        C_adj = C.adjacency_matrix_scipy(0, True, 'csr')
-        C_adj.data = F.asnumpy(C_weights)
-        C_adj = F.tensor(C_adj.todense(), dtype=dtype)
-        c = F.tensor((a * b).todense(), dtype=dtype)
-        assert F.allclose(C_adj, c)
+@pytest.mark.parametrize('dtype', [F.float32, F.float64])
+def test_csrmm(idtype, dtype):
+    a, A = _random_simple_graph(idtype, dtype, F.ctx(), 500, 600, 9000, 'A', 'B', 'AB')
+    b, B = _random_simple_graph(idtype, dtype, F.ctx(), 600, 700, 9000, 'B', 'C', 'BC')
+    C, C_weights = dgl.sparse._csrmm(A._graph, A.edata['w'], B._graph, B.edata['w'], 2)
```

Comment:
Why need we test this internal API? Will we suggest users to use it in the future? In other word, why not directly testing `dgl.adj_product_graph`?